internal/agent/agent.go

```diff
@@ -70,6 +70,7 @@ func VisitWebPage(c *httpbakery.Client, username string) func(u *url.URL) error
 		if err != nil {
 			return errgo.Notef(err, "cannot create login request")
 		}
+		req.Header.Set("Content-Type", "application/json")
```

Comment:
We should probably be using a standard function to make a JSON request here, which would have avoided this bug.